src/Symfony/Component/PropertyInfo/WriteMutator.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * The write mutator defines how a property can be written.
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+final class WriteMutator
+{
+    public const TYPE_METHOD = 1;
+    public const TYPE_PROPERTY = 2;
+    public const TYPE_ADDER_REMOVER_METHOD = 3;
+    public const TYPE_CONSTRUCTOR = 4;
+
+    private $type;
+    private $name;
+    private $private;
+    private $static;
+    private $adder;
+    private $remover;
+
+    private function __construct()
+    {
+    }
+
+    public function getType(): int
+    {
+        return $this->type;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function getAdder(): ?self
+    {
+        return $this->adder;
+    }
+
+    public function getRemover(): ?self
+    {
+        return $this->remover;
+    }
+
+    public function isPrivate(): ?bool
```

Comment:
i'd prefer to throw an exception when the mutator is of a type where this information makes no sense (like constructor)